calyx-py/calyx/builder_util.py

```diff
@@ -0,0 +1,175 @@
+# pylint: disable=import-error
+import calyx.builder as cb
+
+
+def insert_eq(comp: cb.ComponentBuilder, left, right, cellname, width):
+    """Inserts wiring into component {comp} to check if {left} == {right}.
+    1. Within {comp}, creates a combinational group called {cellname}_group.
+    2. Within the group, creates a cell {cellname} that checks equalities of {width}.
+    3. Puts the values {left} and {right} into the cell.
+    4. Returns the equality-checking cell and the overall group.
+    """
+    eq_cell = comp.eq(cellname, width)
+    with comp.comb_group(f"{cellname}_group") as eq_group:
+        eq_cell.left = left
+        eq_cell.right = right
+    return eq_cell, eq_group
+
+
+def insert_lt(comp: cb.ComponentBuilder, left, right, cellname, width):
+    """Inserts wiring into component {comp} to check if {left} < {right}.
+    1. Within {comp}, creates a combinational group called {cellname}_group.
+    2. Within the group, creates a cell {cellname} that checks less-than of {width}.
+    3. Puts the values {left} and {right} into the cell.
+    4. Returns the less-than-checking cell and the overall group.
+    """
+    lt_cell = comp.lt(cellname, width)
+    with comp.comb_group(f"{cellname}_group") as lt_group:
+        lt_cell.left = left
+        lt_cell.right = right
+    return lt_cell, lt_group
```

Comment:
If you want, these similar-looking bodies could be factored out into a helper function (parameterized on the name of the binary-operator component to instantiate).

Reply:
I've taken this advice partially! I've done up a helper function that takes a _cell_ (not a cell name) and makes a combinational group and puts `left` and `right` into the group. This is now in use. I think I'd like to leave it at that for readability, if that's okay?

Reply:
Great!